File: 7-kyu/cryptanalysis-word-patterns/python/solution.py
def word_pattern(word):
    seen_letters = {}
    curr = 0
    ret = []
    for letter in word.lower():
        if letter not in seen_letters:
            seen_letters[letter] = curr
            curr += 1
        ret.append(seen_letters[letter])
    return '.'.join(str(elem) for elem in ret)

File: 7-kyu/cryptanalysis-word-patterns/javascript/solution.js
const wordPattern = word => {
  const seenLetters = new Map();
  let curr = 0;
  const pattern = [];
  for (const letter of word.toLowerCase()) {
    if (!seenLetters.has(letter)) {
      seenLetters.set(letter, curr);
      curr++;
    }
    const currLetter = seenLetters.get(letter);
    pattern.push(currLetter);
  }
  return pattern.map(letter => letter.toString()).join(".")
};
